project/src/main/java/com/google/sps/servlets/AuthServlet.java

```diff
@@ -89,17 +75,75 @@ private String verifyIDToken(HttpServletRequest request) throws IOException{
       buffer.append(line);
       buffer.append(System.lineSeparator());
     }
-    String body = buffer.toString();
+    String idToken = buffer.toString();
+
+    long expiresIn = TimeUnit.DAYS.toMillis(5);
+    SessionCookieOptions options = SessionCookieOptions.builder()
+        .setExpiresIn(expiresIn)
+        .build();
 
-    //decode and verify client ID token
-    FirebaseToken decodedToken = null;
     try {
-      decodedToken = FirebaseAuth.getInstance().verifyIdToken(body);
+      // Create the session cookie. This will also verify the ID token in the process.
+      // The session cookie will have the same claims as the ID token.
+      String sessionCookie = FirebaseAuth.getInstance().createSessionCookie(idToken, options);
+      Cookie firebaseCookie = new Cookie("firebase_session", sessionCookie);
+      response.addCookie(firebaseCookie);
+
+      FirebaseToken decodedToken = FirebaseAuth.getInstance().verifyIdToken(idToken);
+      if (decodedToken == null) {
+        response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+        response.getWriter().write("Invalid token");
+        response.getWriter().flush();
+        return;
+      }
+
+      String email = decodedToken.getEmail();
+      long userID = getUserID(request,email);
+      request.getSession().setAttribute("userID", userID);
+
+      response.setStatus(HttpServletResponse.SC_OK);
+      response.flushBuffer();
     } catch (FirebaseAuthException e) {
-      e.printStackTrace();
+      response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+      response.getWriter().write("Failed to create a session cookie");
+      response.getWriter().flush();
     }
+  }
+
+  @Override
+  public void doDelete(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    request.getSession().setAttribute("userID", null);
 
-    return decodedToken.getUid();
+    Cookie emptyCookie = new Cookie("firebase_session", null);
+    response.addCookie(emptyCookie);
+    response.setStatus(HttpServletResponse.SC_OK);
+    response.flushBuffer();
+  }
+
+  /**
+   * If user is in the database, get the user ID in the database from the given email.
+   * Otherwise add user to the database and return the user's ID.
+   * @param request -- HTTP request
+   * @param email -- user's email from the decoded token
+   * @return -- user's ID in the database
+   */
+  private long getUserID(HttpServletRequest request, String email){
```

Comment:
I think this function name should convey that a user will also be created if not found in the database - eg `getOrCreateUserID`